.buildkite/pipeline.json.py

```diff
@@ -48,8 +47,10 @@ def main():
     if config.build_linux:
         build_linux = pipeline_steps.generate_step_template("Linux", config.action, config.snapshot, config.version_qualifier)
         pipeline_steps.append(build_linux)
-    pipeline_steps.append(pipeline_steps.generate_step("Upload ES tests runner pipeline",
-                                                       ".buildkite/pipelines/run_es_tests.yml.sh"))
+    # Never run the ES integration tests if VERSION_QUALIFIER has been set.
```

Comment:
I can see why this won't work (it would actually if we passed the same qualifier through to the ES build) but it was never a problem before because:

1. Nobody ever used version qualifiers on PR builds
2. Release branch builds never ran the ES integration tests

I think it's best to preserve the old behaviour of release branch builds not running the ES tests. In PR builds we want to catch incompatibilities between the C++ and Java that have been accidentally introduced. In the release branch builds we're often under pressure to produce artifacts for a new version. We don't want these branch builds to fail because Elasticsearch is at a different stage of its version bumping process to ML.